core/arch/arm/sm/pm_a32.S

```diff
@@ -104,14 +104,16 @@ UNWIND(	.cantunwind)
 	/*
 	 * At this point, MMU is not enabled now.
 	 * 1. Get the runtime physical address of _suspend_sp
-	 * 2. Get the offset from _suspend_sp to &thread_core_local
+	 * 2. Get the offset from _suspend_sp to thread_core_local
 	 * 3. Get the runtime physical address of thread_core_local
 	 * Since moving towards non-linear mapping,
-	 * `ldr r0, =thread_core_local` is not used here.
+	 * `ldr r0, =thread_core_local
+	 *  ldr r0 [r0]` is not used here.
 	 */
 	adr	r4, _suspend_sp
 	ldr	r5, [r4]
 	add	r4, r4, r5
+	ldr	r4, [r5]
```

Comment:
something seems wrong here: shouldn't it be `ld r4, [r4]`?

Reply:
Yes, you're right.